src/cephadm/cephadm

```diff
@@ -2198,6 +2200,14 @@ def command_shell():
     if daemon_id and not args.fsid:
         raise Error('must pass --fsid to specify cluster')
 
+    # use /etc/ceph files by default, if present.  we do this instead of
+    # making these defaults in the arg parser because we don't want an error
+    # if they don't exist.
+    if not args.config and os.path.exists(SHELL_DEFAULT_CONF):
```

Comment:
`--config` and `--keyring` are also available in `cephadm deploy` command. Is it intentional to only have this default behavior in `cephadm shell` command?

Reply:
Yes.  ``shell`` is consumed by a human who doesn't want to type.  ``deploy`` is consumed by mgr/cephadm and is always passed the config and keyring for the specific daemon being deployed